packages/framework/src/client/scheduler.js

```diff
@@ -0,0 +1,140 @@
+import { find } from 'lodash';
+import ResourceRequest, { STATUS } from './resource-request';
+
+const DEFAULT_FETCH_INTERVAL = 5000; // Five seconds
+
+export default class Scheduler {
+	constructor(
+		fetch,
+		fetchInterval = DEFAULT_FETCH_INTERVAL,
+		setInterval = window.setInterval,
+	) {
+		this.fetch = fetch;
+		this.requests = [];
+		this.timerId = null;
+
+		if ( fetchInterval > 0 ) {
+			setInterval( () => {
+				this.cleanUp();
+				this.resendTimeouts();
+				this.sendReadyRequests();
+			}, fetchInterval );
+		}
+	}
+
+	/**
+	 * Finds a request that is either scheduled or overdue.
+	 * @param {string} resourceName The name of the resource to be read
+	 * @return {ResourceRequest} The scheduled request, or null if none found
+	 */
+	getScheduledRequest = ( resourceName ) => {
+		return find( this.requests, ( r ) => {
+			if ( resourceName === r.resourceName ) {
+				const status = r.getStatus();
+				return STATUS.scheduled === status || STATUS.overdue === status;
+			}
+			return false;
+		} );
+	}
+
+	/**
+	 * Finds requests that are in flight by resourceName
+	 * @param {string} resourceName The name of the resource to check
+	 * @return {Array} Any requests for the given resource which are currently in flight.
```

Comment:
Why would there be more than 1 in-flight request for a given resource?

Reply:
For a read, this shouldn't ever happen. But in Step 3, support for other operations is introduced, which have data payloads that could be different.